src/layout/Summary2/SummaryComponent2/ComponentSummary.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+
+import { Grid } from '@material-ui/core';
+import cn from 'classnames';
+
+import classes from 'src/layout/Summary2/SummaryComponent2/SummaryComponent2.module.css';
+import { gridBreakpoints, pageBreakStyles } from 'src/utils/formComponentUtils';
+import { useNode } from 'src/utils/layout/NodesContext';
+import type { CompSummary2External, CompSummary2Internal } from 'src/layout/Summary2/config.generated';
+import type { LayoutNode } from 'src/utils/layout/LayoutNode';
+
+interface ComponentSummaryProps {
+  componentNode: LayoutNode;
+  summaryOverrides: CompSummary2Internal['overWriteProperties'];
+}
+
+interface ResolveComponentProps {
+  summaryProps: CompSummary2External;
+  summaryOverrides: any;
+}
+
+export function ComponentSummary({ componentNode, summaryOverrides }: ComponentSummaryProps) {
+  if (componentNode.isHidden()) {
+    return null;
+  }
+
+  const overrides = summaryOverrides?.find((override) => override.componentId === componentNode.item.id);
+
+  const childComponents =
+    componentNode.item.type === 'Group' &&
+    componentNode.item.childComponents.map((child) => (
+      <ComponentSummary
+        componentNode={child}
+        key={child.item.id}
+        summaryOverrides={summaryOverrides}
+      />
+    ));
```

Comment:
Groups are far from the only components with children nowadays. Instead of reaching directly into the node state, why not use node.children()?

Reply:
Yes, this needs work, will look at it tomorrow :)